skills_ml/algorithms/preprocessing/__init__.py

```diff
@@ -0,0 +1,82 @@
+from functools import reduce
+
+from skills_ml.algorithms.string_cleaners.nlp import NLPTransforms, deep
+
+
+class NLPPipeline(object):
```

Comment:
How about `IterablePipeline`? Not only is this class generic enough to not need to be NLP-related, but I think the fact that it expects to deal with iterables is relevant and useful info. I was tempted to say `GeneratorPipeline`, but that's not really true: you could pass in a fully materialized list and it would be fine.

Reply:
That's true. IterablePipeline sounds good to me.